demos/projects/ST/b-l4s5i-iot01a/README.md

```diff
@@ -0,0 +1,112 @@
+
+*While this sample supports SAS keys and Certificates, this guide will refer only to SAS keys. 
+
+### Install prerequisites on Windows
+
+Ensure that cmake, ninja and the ARM toolset binaries are available in the `PATH` environment variable.
+
+You may also need to enable long path support for both Windows and git:
+* Windows: <https://docs.microsoft.com/windows/win32/fileio/maximum-file-path-limitation?tabs=cmd#enable-long-paths-in-windows-10-version-1607-and-later>
+* Git: As **administrator** run `git config --system core.longpaths true`
+
+## Get the middleware
+
+Clone the following repo to download all sample device code, setup scripts, and offline versions of the documentation. 
+
+*If you previously cloned this repo in another sample, you don't need to do it again.*
+
+```bash
+git clone https://github.com/Azure-Samples/iot-middleware-freertos-samples.git
+```
+
+To initialize the repo, run the following command:
+
+```bash
+git submodule update --init --recursive --depth 1
+```
+
+## Prepare the device
+To connect the STM DevKit to Azure, you'll modify a configuration file for Wi-Fi and Azure IoT settings, rebuild the image, and flash the image to the device.
+
+Update the file `iot-middleware-freertos-samples/demos/projects/ST/b-l4s5i-iot01a/config/demo_config.h` with your configuration values.
+
+Set the **Wi-Fi parameters** to the following values for your local environment.
+
+Parameter | Value 
+---------|----------
+ `WIFI_SSID` | _{Your Wi-Fi SSID}_
+ `WIFI_PASSWORD` | _{Your Wi-Fi Password}_ 
+
+If you're using a device previously created in your **IoT Hub** with SAS authentication, disable DPS by commeting out line #53 (`#define democonfigENABLE_DPS_SAMPLE`) and set the following parameters:
+
+Parameter | Value 
+---------|----------
+ `democonfigDEVICE_ID` | _{Your Device ID value}_
+ `democonfigHOSTNAME` | _{Your Azure IoT Hub Host name value}_ 
+ `democonfigDEVICE_SYMMETRIC_KEY` | _{Your Primary Key value}_ 
+
+If you're using **DPS** with an individual enrollment with SAS authentication, set the following parameters:
+
+Parameter | Value 
+---------|----------
+ `democonfigID_SCOPE` | _{Your DPS ID scope value}_
+ `democonfigREGISTRATION_ID` | _{Your Device Registration ID value}_ 
+ `democonfigDEVICE_SYMMETRIC_KEY` | _{Your Primary Key value}_ 
+
+## Build the image
+
+To build the device image, navigate to the `iot-middleware-freertos-samples` directory and run the following commands:
+
+  ```bash
+    cmake -G Ninja -DVENDOR=ST -DBOARD=b-l4s5i-iot01a -Bb-l4s5i-iot01a .
+    cmake --build b-l4s5i-iot01a
+  ```
+
+After the build completes, confirm that a folder named `/b-l4s5i-iot01a/` was created and it contains a file named `iot-middleware-sample.bin`. 
```

Comment:
this might be buried underneath some subdirectories right? like demos/projects/blah/blah. Should we call that out since it's not actually at the root of the build?

Reply:
this need to be fixed in all the readme... will address in next PR.